src/ReverseProxy/Service/RuntimeModel/Transforms/QueryTransformContext.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.ReverseProxy.Service.RuntimeModel.Transforms
+{
+    /// <summary>
+    /// Transform state for use with <see cref="RequestParametersTransform"/>
+    /// </summary>
+    public class QueryTransformContext
+    {
+        private readonly HttpRequest _request;
+        private readonly QueryString _originalQueryString;
+        private Dictionary<string, StringValues> _modifiedQueryParameters;
+
+        public QueryTransformContext(HttpRequest request)
+        {
+            _request = request ?? throw new ArgumentNullException(nameof(request));
+            _originalQueryString = request.QueryString;
+            _modifiedQueryParameters = null;
+        }
+
+        public QueryString QueryString
+        {
+            get
+            {
+                if (_modifiedQueryParameters == null)
+                {
+                    return _originalQueryString;
+                }
+
+#if NETCOREAPP3_1
+                var queryBuilder = new QueryBuilder(_modifiedQueryParameters.Select(pair => new System.Collections.Generic.KeyValuePair<string, string>(pair.Key, pair.Value.ToString())));
+#elif NETCOREAPP5_0
+                var queryBuilder = new QueryBuilder(_modifiedQueryParameters);
```

Comment:
I think there's an SDK bug. Try swapping the order, and switch to the new TFM:
```suggestion
#if NET5_0
                var queryBuilder = new QueryBuilder(_modifiedQueryParameters);
#elif NETCOREAPP3_1
                var queryBuilder = new QueryBuilder(_modifiedQueryParameters.Select(pair => new System.Collections.Generic.KeyValuePair<string, string>(pair.Key, pair.Value.ToString())));
```

Reply:
https://github.com/dotnet/runtime/issues/41839